src/dataLoader/utils.spec.ts

```diff
@@ -25,6 +25,24 @@ describe('reorder by key', () => {
     const actual = reorderResultByKey<MyObject, 'name'>(orderMapping, results);
     expect(actual).toEqual(expected);
   });
+  it('undefined returned when key is not found', () => {
+    const results: MyObject[] = [
+      {
+        id: 1,
+        name: '1',
+        next: { id: 1, name: '2', next: null },
+      },
+      {
+        id: 2,
+        name: '2',
+        next: null,
+      },
+    ];
+    const expected = [undefined, results[0]];
```

Comment:
utility function does the mapping for the input given to it..decided not to filter out `undefined` here as it can be handled by calling function if required.